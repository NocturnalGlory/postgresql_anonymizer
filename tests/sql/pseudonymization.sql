BEGIN;

CREATE EXTENSION IF NOT EXISTS anon CASCADE;

SELECT anon.load();

-- First Name
SELECT  anon.pseudo_first_name('bob')
      = anon.pseudo_first_name('bob');

SELECT  anon.pseudo_first_name('bob','123salt*!')
      = anon.pseudo_first_name('bob','123salt*!');

SELECT anon.pseudo_first_name(NULL) IS NULL;

SELECT pg_typeof(anon.pseudo_first_name(NULL)) = 'TEXT'::REGTYPE;

ROLLBACK;
