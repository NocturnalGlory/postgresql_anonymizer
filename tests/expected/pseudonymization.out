BEGIN;
CREATE EXTENSION IF NOT EXISTS anon CASCADE;
SELECT anon.load();
 load 
------
 t
(1 row)

-- First Name
SELECT  anon.pseudo_first_name('bob')
      = anon.pseudo_first_name('bob');
 ?column? 
----------
 t
(1 row)

SELECT  anon.pseudo_first_name('bob','123salt*!')
      = anon.pseudo_first_name('bob','123salt*!');
 ?column? 
----------
 t
(1 row)

SELECT anon.pseudo_first_name(NULL) IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.pseudo_first_name(NULL)) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

ROLLBACK;
